tests/server_spec.rs

```diff
@@ -0,0 +1,38 @@
+use tailcall::config::Config;
+use tailcall::http::start_server;
+
+#[tokio::test]
+async fn server_start() {
+  use reqwest;
+  use reqwest::Client;
+  use serde_json::json;
+
+  tokio::spawn(async {
+    let file_paths = vec!["tests/server/config/server-start.graphql".to_string()];
+    let config = Config::from_file_or_url(file_paths.iter()).await.unwrap();
+    start_server(config).await.unwrap();
+  });
```

Comment:
According to [tokio docs](https://tokio.rs/tokio/tutorial/spawning#tasks), we can 
```suggestion
  tokio::spawn(async {
    let file_paths = vec!["tests/server/config/server-start.graphql".to_string()];
    let config = Config::from_file_or_url(file_paths.iter()).await.unwrap();
    start_server(config).await.unwrap()
  }).await.unwrap();
```

In this way we don't need to loop until the server starts

Reply:
Like I mentioned before this results in infinite loading

Reply:
@meskill @tusharmath @amitksingh1490, guys, can you think about an alternative here?

I've also tried to save the spawn in a variable and then loop until `spawn.is_finished()` became true, but this does not work either.

Reply:
I would recommend using a `oneshot::channel()` to perform this synchronization. `start_server` should take a `oneshot` and set the value once the handler is registered. In the test we should await on the reader.